src/plugins/URLIndicatorPlugin/URLIndicator.js

```diff
@@ -0,0 +1,103 @@
+        // DISCONNECTED: HTTP failed; maybe misconfigured, disconnected.
+        // PENDING: Still trying to connect, and haven't failed yet.
+        // ERROR: Got answer but it's due an internal server error or similar.
+        var CONNECTED = {
+                text: "connected",
+                glyphClass: "ok",
+                description: "Connected to the URL"
+            },
+            DISCONNECTED = {
+                text: "disconnected",
+                glyphClass: undefined,
+                description: "URL is offline"
+            },
+            PENDING = {
+                text: "Checking connection...",
+                glyphClass: 'caution',
+                description: "Unable to connect to the URL"
+            },
+            ERROR = {
+                text: "Error",
+                glyphClass: "err",
+                description: "Error trying to connect to the URL"
+            };
+        function URLIndicator($http, $interval) {
+            var self = this;
+            // Get the current connection state and options provided
+            this.icon = "icon-" + (this.options.icon ? this.options.icon : "database");
+            this.URLpath = this.options.url;
+            this.interval = this.options.interval || 1000;
+            this.state = PENDING;
+            if (this.options.label) {
+                this.label = this.options.label;
+            } else {
+                this.label = "URL";
+            }
+            // Callback if the HTTP request to URLIndicatorPlugin fails
+            function handleError(e) {
+                if (e.status !== 404) {
+                    self.state = ERROR;
+                } else {
+                    self.state = DISCONNECTED;
+                }
+            }
+            // Callback if the HTTP request succeeds.
+            function handleResponse() {
+                self.state = CONNECTED;
+            }
+            // Try to connect to the URL, and update the indicator.
+            function updateIndicator(path) {
+                path = self.URLpath;
+                $http.get(path).then(handleResponse, handleError);
+            }
+            // Update the indicator initially, and start polling.
+            updateIndicator(this.URLpath);
+            $interval(updateIndicator, this.interval, 0, false);
+        }
+
+        URLIndicator.prototype.getCssClass = function () {
+            return this.icon;
+        };
+        URLIndicator.prototype.getGlyphClass = function () {
+            return this.state.glyphClass;
+        };
+        URLIndicator.prototype.getText = function () {
+            if (this.state.text === "connected" || this.state.text === "disconnected") {
+                return this.label + " is " + this.state.text;
+            }
+            return this.state.text;
+        };
+        URLIndicator.prototype.getDescription = function () {
+            return this.state.description;
```

Comment:
Description should be customized based on option and status:

pending: 'Checking status of <label>, stand by...'
connected: '<label> is online, checking status every <interval> milliseconds.'
disconnected: '<label> is offline, checking status every <interval> milliseconds.'